adapters/sharethrough/butler.go

```diff
@@ -55,6 +55,8 @@ func (s StrOpenRTBTranslator) requestFromOpenRTB(imp openrtb.Imp, request *openr
 	headers.Add("Content-Type", "text/plain;charset=utf-8")
 	headers.Add("Accept", "application/json")
 	headers.Add("Origin", domain)
+	headers.Add("X-Forwarded-For", request.Device.IP)
```

Comment:
What will happen here if `request.Device == nil`?

Reply:
Is that possible? in which cases? 🤔 the adapter already relies on the `Device` even prior to this PR, eg. parsing the User Agent

Reply:
Went back into the code and saw that we do try setting `Device` implicitly if it does not exist. However if we cannot deduce the IP nor the user agent from a request, and `device` is omitted from the request, then `Device` will be `nil`. I am not sure it is possible to fail to pull the IP from an http request, if all else fails we use `net.SplitHostPort(r.RemoteAddr)`, as long as this does not return `""` for the IP then `Device` will exist.

So we are probably actually good here.

Reply:
Cool, thanks for sharing context too 👍